src/jit/codegenarm.cpp

```diff
@@ -294,7 +295,14 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             // The src must be a register.
             regNumber operandReg = genConsumeReg(operand);
 
-            getEmitter()->emitIns_R_R_I(ins, emitTypeSize(treeNode), targetReg, operandReg, 0);
+            if (ins == INS_vneg)
+            {
+                getEmitter()->emitIns_R_R(ins, emitTypeSize(treeNode), targetReg, operandReg);
+            }
```

Comment:
@sivarv I've faced the case where GT_NEG has float type when running `JIT/CodeGenBringUpTests/FPRoots/FPRoots.exe`. That's the case I've describe in the PR description as "A new assertion occurs at codegen for GT_NET of floating point type.".

At line 286 (several lines above) `instruction ins = genGetInsForOper(treeNode->OperGet(), targetType);`, genGetInsForOper() checks `varTypeIsFloating(type)` and returns `INS_vneg` for float type.
If `GT_NEG` should not have float type, I think there is some problem in former phases including import phase because I've observed `unary -   float` was generated as below.

```
*************** In impImport() for BringUpTest:FPRoots(float,float,float,byref,byref)
               ...
               [000039] -------------             |  |  \--*  -         float
               [000026] -------------             |  |     \--*  unary -   float
               [000025] -------------             |  |        \--*  lclVar    float  V01 arg1
```
What do you think of it ?
If `GT_NEG` should not have float/double type. I'm going to look into front-end of ryujit/arm32 too.